packages/apputils/src/dialog.tsx

```diff
@@ -436,12 +436,14 @@ export class Dialog<T> extends Widget {
         event.preventDefault();
 
         const activeEl = document.activeElement;
-        let index: number | undefined;
+        let index: number | undefined = this._defaultButton;
 
         if (activeEl instanceof HTMLButtonElement) {
           index = this._buttonNodes.indexOf(activeEl);
+          this.resolve(index !== -1 ? index : 0);
+        } else if (!(activeEl instanceof HTMLTextAreaElement)) {
+          this.resolve(index !== undefined ? index : 0);
```

Comment:
Index here is always `this._defaultButton` which is never undefined. Against, we can simplify the code:

```suggestion
          const index = this._defaultButton;
          this.resolve(index);
```